esp-wifi/src/wifi/mod.rs

```diff
@@ -340,6 +343,169 @@ impl Default for ClientConfiguration {
     }
 }
 
+#[allow(clippy::type_complexity)]
+static CSI_CB: Mutex<RefCell<Option<fn(crate::binary::include::wifi_csi_info_t)>>> =
+    Mutex::new(RefCell::new(None));
+
+unsafe extern "C" fn promiscuous_csi_rx_cb(
+    _ctx: *mut crate::wifi::c_types::c_void,
+    data: *mut crate::binary::include::wifi_csi_info_t,
+) {
+    critical_section::with(|cs| {
+        let Some(csi_callback) = *CSI_CB.borrow_ref(cs) else {
+            return;
+        };
+        csi_callback(*data);
+    });
+}
+
+#[derive(Clone, PartialEq, Eq)]
+// https://github.com/esp-rs/esp-wifi-sys/blob/main/esp-wifi-sys/headers/local/esp_wifi_types_native.h#L94
+/// Channel state information(CSI) configuration
+#[cfg(not(esp32c6))]
+pub struct CsiConfiguration {
```

Comment:
Probably this should be `CsiConfig`, as the other configuration struct(s) use `Config`.